File: applications/backend/api/src/main/kotlin/live/forseti/api/adapter/dto/response/session/SessionResponseDTO.kt
package live.forseti.api.adapter.dto.response.session

import live.forseti.api.adapter.dto.response.member.MemberPublicResponseDTO
import live.forseti.api.adapter.dto.response.member.toPublicResponseDTO
import live.forseti.core.domain.entity.Session
import java.time.OffsetDateTime
import java.util.UUID

data class SessionResponseDTO(
    val id: UUID,
    val member: MemberPublicResponseDTO,
    val expiresAt: OffsetDateTime,
)

fun Session.toResponseDTO() =
    SessionResponseDTO(
        id = id,
        member = member.toPublicResponseDTO(),
        expiresAt = expiresAt,
    )

File: applications/webapp/src/core/port/dto/response/session/SessionResponseDTO.ts
import { MemberResponseDTO } from "@/core/port/dto/response/member/MemberResponseDTO";

export type SessionResponseDTO = {
  id: string;
  createdAt: string;
  updatedAt: string;
  member: MemberResponseDTO;
  expiresAt: string;
  version: number;
};
